services/spar/schema/src/V0.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes       #-}
+
+module V0 (migration) where
+
+import Cassandra.Schema
+import Control.Monad (void)
+import Text.RawString.QQ
+
+migration :: Migration
+migration = Migration 0 "Initial schema" $ do
+
+    -- TODO: review authreq/authresp/user tables; `text` likely non-unique and thus a problematic
+    --       primary key.  see existing schema for inspiration:
+    --       https://github.com/wireapp/wire-server/blob/312f82344fa153dad76a03d7280b4424174c2fac/doc/db_schema.cql
+
+    -- TODO: on read-heavy columns, use
+    --       with compaction = {'class': 'LeveledCompactionStrategy'};
+    --       see https://www.datastax.com/dev/blog/when-to-use-leveled-compaction
+
+    -- FUTUREWORK: in authreq, field req, we may be able to use UUID, because we can create those?
+    void $ schema' [r|
+        CREATE TABLE if not exists authreq
+            ( req          text
+            , end_of_life  timestamp
+            , primary key  (req)
+            );
+        |]
+
+    void $ schema' [r|
+        CREATE TABLE if not exists authresp
+            ( resp         text
+            , end_of_life  timestamp
+            , primary key  (resp)
+            );
+        |]
+
+    void $ schema' [r|
+        CREATE TABLE if not exists user
+            ( idp      uuid
+            , sso_id   text
+            , uid      uuid
+            , primary key (idp, sso_id)
+            );
+        |]
+
+    void $ schema' [r|
+        CREATE TABLE if not exists idp
+            ( idp           uuid
+            , metadata      text
+            , issuer        text
+            , request_uri   text
+            , public_key    text
```

Comment:
Let's make `public_key` a blob instead? [service](https://github.com/wireapp/wire-server/blob/develop/services/brig/schema/src/V35.hs#L31-L54) is an interesting one to look at

Reply:
37ec78c337a53f5618dfecf64a4c947335811bfb (will push soon)

(I also found another oversight in the `fail` message.)